### ISSUE TITLE
Unicode decoding issue when loading file from GitHub

### ISSUE THREAD

**Issue by ronnix** [mapbox/geojson.io#459]
When uploading geoJSON file from my computer, properties that include non-ASCII Unicode characters display correctly (e.g. "Café").

When loading the exact same file from GitHub, properties are not displayed correctly (e.g. "CafÃ©").

Note that this happens whether the file is UTF-8 encoded, or if I use `\u00e9` notation for non-ASCII characters.

**Comment by jeekajoo** [comment]
Same here.
https://github.com/ParisEnSelle/planvelo-carte/blob/master/planvelo.geojson
https://github.com/ParisEnSelle/planvelo-carte/blob/master/planvelo_modified.geojson

Diff extract
```
311c311
<         "name": "Place des Fêtes"
---
>         "name": "Place des FÃªtes"
```

**Comment by jeekajoo** [comment]
```bash
$ file planvelo*
planvelo.geojson:          UTF-8 Unicode text
planvelo_modified.geojson: UTF-8 Unicode text
```
@tmcw

**Comment by Loufylouf** [comment]
This is apparently a well known problem that is documented here : https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem . I tried the first solution which works perfectly, though I'm not sure that it is the most efficient. 

If you wish to integrate it, I suggest modifying src/ui/file_bar.js (lines 290-298) from 
```
                    if (last.type === 'blob') {
                        githubBrowser.request('/repos/' + d[1].full_name +
                            '/git/blobs/' + last.sha, function(err, blob) {
                                d.content = JSON.parse(atob(blob[0].content));
                                context.data.parse(d);
                                zoomextent(context);
                                m.close();
                            });
                    }
```
to : 
```
                    if (last.type === 'blob') {
                        githubBrowser.request('/repos/' + d[1].full_name +
                            '/git/blobs/' + last.sha, function(err, blob) {
                                d.content = JSON.parse(decodeURIComponent(Array.prototype.map.call(atob(blob[0].content), function(c) { return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2); }).join('')));
                                context.data.parse(d);
                                zoomextent(context);
                                m.close();
                            });
                    }
```
It's a quick & dirty fix but I'm not familiar enough with geojson.io to know where to put the b64DecodeUnicode function definition.

**Comment by mykolakozyr** [comment]
As I understand it cannot be implemented just by forking the repository and changing 'file_bar.js', right?
I have to fork the repository, create my own gatekeeper, change config.js, rebuild geojson.io and be happy (based on [this comment](https://github.com/mapbox/geojson.io/issues/366#issuecomment-49694126) )?

**Comment by tmcw** [comment]
Hey Mykola,

That's only necessary if you want to run your own geojson.io. This ^ bug isn't something that'd necessitate that; if someone wants to submit a pull request, then I'd accept it and the bug would be fixed for everyone, which is much better than just one person getting the fix.

Thanks,

- Tom

**Comment by tmcw** [comment]
This is fixed in #459

**Comment by ronnix** [comment]
Thanks, looks like it works well now!
